### DISCUSSION TITLE
Example for using pytorch_geometric on .STL Data (CAD)

### DISCUSSION THREAD

**Discussion by Boehm92** [pyg-team/pytorch_geometric#4092] [Q&A] [answered]
Hi,

I'm new to pytorch_geometric and would like to use it for my Ph.D. I wrote a code to convert a CAD file in .STL format (MESH) into a Data Object. Right now I import a simple Cube with numpy-stl and try to convert it. The .STL format for the bottom faces looks like that:
```
 facet normal -1.000000e+00 0.000000e+00 0.000000e+00
      outer loop
         vertex 0.000000e+00 0.000000e+00 1.000000e+02
         vertex 0.000000e+00 1.000000e+02 1.000000e+02
         vertex 0.000000e+00 0.000000e+00 0.000000e+00
      endloop
   endfacet
   facet normal -1.000000e+00 0.000000e+00 0.000000e+00
      outer loop
         vertex 0.000000e+00 0.000000e+00 0.000000e+00
         vertex 0.000000e+00 1.000000e+02 1.000000e+02
         vertex 0.000000e+00 1.000000e+02 0.000000e+00
      endloop
   endfacet
```

If I plot the converted data object with the plot example from Antonio Longa I get this graph:
![image](https://user-images.githubusercontent.com/40231115/154424379-7be40b29-e296-4322-8185-74e68313f3ad.png)

The edge_index list seems correct, only the nodes are distorted. Is there already an example to convert .STL files correctly or is the problem with the to_network() method maybe?

I would be grateful for the help.
Kind Regards
Stefan

**Comment by rusty1s** [comment, marked as answer]
That looks indeed a bit wrong, but I doubt that this relates to a `to_networkx()` bug. Have you also looked at the visualization tools used [here](https://colab.research.google.com/drive/1D45E5bUK3gQ40YpZo65ozs7hg5l-eo_U?usp=sharing)?

**Reply by Boehm92** [reply]
Thanks, didnt check this tutorial out. It seems like that I have some bug creating the edge_index list
